src/test/java/org/elasticsearch/index/query/CommonTermsQueryBuilderTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.base.Joiner;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.queries.ExtendedCommonTermsQuery;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.mapper.FieldMapper;
+import org.elasticsearch.index.query.CommonTermsQueryBuilder.Operator;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class CommonTermsQueryBuilderTest extends BaseQueryTestCase<CommonTermsQueryBuilder> {
+
+    @Override
+    protected CommonTermsQueryBuilder createTestQueryBuilder() {
+        CommonTermsQueryBuilder query;
+        
+        // mapped or unmapped field
+        String text = "text" + "_" + Joiner.on(" ").join(generateRandomStringArray(10, 10, false));
+        if (randomBoolean()) {
+            query = new CommonTermsQueryBuilder(STRING_FIELD_NAME, text);
+        } else {
+            query = new CommonTermsQueryBuilder(STRING_FIELD_NAME + "_"  + randomAsciiOfLengthBetween(1, 10), text);
+        }
+        
+        // random cutoff frequency
+        query.cutoffFrequency(randomIntBetween(1, 10));
+        
+        // low frequency terms operator
+        query.lowFreqOperator(randomFrom(Operator.values()));
+        
+        // if or, number of low frequency terms that must match
+        if (query.lowFreqOperator().equals(Operator.OR)) {
+            query.lowFreqMinimumShouldMatch("" + randomIntBetween(1, 5));
+        }
+
+        // high frequency terms operator
+        query.highFreqOperator(randomFrom(Operator.values()));
+
+        // if or, number of high frequency terms that must match
+        if (query.highFreqOperator().equals(Operator.OR)) {
+            query.highFreqMinimumShouldMatch("" + randomIntBetween(1, 5));
+        }
```

Comment:
we never test defaults I'm afraid when it comes to `cutoffFrequency`, `lowFreqOperator`, `lowFreqMinimumShouldMatch`, `highFreqOperator`, `highFreqMinimumShouldMatch`?